src/Persisters/Entity/JoinedSubclassPersister.php

```diff
@@ -61,8 +61,8 @@ protected function getDiscriminatorColumnTableName(): string
      */
     private function getVersionedClassMetadata(): ClassMetadata
     {
-        if (isset($this->class->fieldMappings[$this->class->versionField]->inherited)) {
-            $definingClassName = $this->class->fieldMappings[$this->class->versionField]->inherited;
+        if (isset($this->class->fieldMappings[$this->class->versionField ?? '']->inherited)) {
```

Comment:
I think it will be clearer to handle the (common) case of classes without a version field explicitly instead of attempting an array check with the empty string.

```suggestion
        if ($this->class->versionField !== null && isset($this->class->fieldMappings[$this->class->versionField]->inherited)) {
```

Reply:
done